src/resource/solidDataset.test.ts

```diff
@@ -915,6 +917,432 @@ describe("saveSolidDatasetAt", () => {
+          },
+        })
+      )
+    );
+
+    const solidDataset = await createEmptyContainerAt(
+      "https://arbitrary.pod/container/",
+      { fetch: mockFetch }
+    );
+
+    expect(solidDataset.internal_resourceInfo.permissions).toEqual({
+      user: {
+        read: true,
+        append: true,
+        write: true,
+        control: true,
+      },
+      public: {
+        read: true,
+        append: false,
+        write: false,
+        control: false,
+      },
+    });
+  });
+
+  it("defaults permissions to false if they are not set, or are set with invalid syntax", async () => {
+    const mockFetch = jest.fn(window.fetch).mockReturnValue(
+      Promise.resolve(
+        new Response(undefined, {
+          headers: {
+            // Public permissions are missing double quotes, user permissions are absent:
+            "WAC-Allow": "public=read",
+          },
+        })
+      )
+    );
+
+    const solidDataset = await createEmptyContainerAt(
+      "https://arbitrary.pod/container/",
+      { fetch: mockFetch }
+    );
+
+    expect(solidDataset.internal_resourceInfo.permissions).toEqual({
+      user: {
+        read: false,
+        append: false,
+        write: false,
+        control: false,
+      },
+      public: {
+        read: false,
+        append: false,
+        write: false,
+        control: false,
+      },
+    });
+  });
+
+  it("does not provide the resource's access permissions if not provided by the server", async () => {
+    const mockFetch = jest.fn(window.fetch).mockReturnValue(
+      Promise.resolve(
+        new Response(undefined, {
+          headers: {},
+        })
+      )
+    );
+
+    const solidDataset = await createEmptyContainerAt(
+      "https://arbitrary.pod/container/",
+      { fetch: mockFetch }
+    );
+
+    expect(solidDataset.internal_resourceInfo.permissions).toBeUndefined();
+  });
+
+  it("returns an empty SolidDataset", async () => {
+    const mockFetch = jest
+      .fn(window.fetch)
+      .mockReturnValue(
+        Promise.resolve(
+          mockResponse(undefined, { url: "https://arbitrary.pod/container/" })
+        )
+      );
+
+    const solidDataset = await createEmptyContainerAt(
+      "https://arbitrary.pod/container/",
+      {
+        fetch: mockFetch,
+      }
+    );
+
+    expect(solidDataset.size).toBe(0);
+  });
+
+  it("returns a meaningful error when the server returns a 403", async () => {
+    const mockFetch = jest
+      .fn(window.fetch)
+      .mockReturnValue(
+        Promise.resolve(new Response("Not allowed", { status: 403 }))
+      );
+
+    const fetchPromise = createEmptyContainerAt(
+      "https://arbitrary.pod/container/",
+      {
+        fetch: mockFetch,
+      }
+    );
+
+    await expect(fetchPromise).rejects.toThrow(
+      new Error("Creating the empty Container failed: 403 Forbidden.")
+    );
+  });
+
+  it("returns a meaningful error when the server returns a 404", async () => {
```

Comment:
Can a `PUT` return 404 ?

Reply:
Ehm no, so this test doesn't really make sense. I'll remove it (it really is redundant with the 403 one anyway).